doc/man3/SSL_shutdown.pod

```diff
@@ -35,35 +35,50 @@ performed, so that the peers stay synchronized.
 SSL_shutdown() supports both uni- and bidirectional shutdown by its 2 step
 behaviour.
 
-=over 4
+SSL_shutdown() only closes the communication in 1 direction.
+It is not possible to call SSL_write() after calling SSL_shutdown().
+
+=head2 First to close the connection
 
-=item When the application is the first party to send the "close notify"
+When the application is the first party to send the "close notify"
 alert, SSL_shutdown() will only send the alert and then set the
 SSL_SENT_SHUTDOWN flag (so that the session is considered good and will
-be kept in cache). SSL_shutdown() will then return with 0. If a unidirectional
-shutdown is enough (the underlying connection shall be closed anyway), this
-first call to SSL_shutdown() is sufficient. In order to complete the
-bidirectional shutdown handshake, SSL_shutdown() must be called again.
-The second call will make SSL_shutdown() wait for the peer's "close notify"
-shutdown alert. On success, the second call to SSL_shutdown() will return
-with 1.
-
-=item If the peer already sent the "close notify" alert B<and> it was
+be kept in the cache).
+SSL_shutdown() will then return with 0.
+If a unidirectional shutdown is enough (the underlying connection shall be
+closed anyway), this first call to SSL_shutdown() is sufficient.
+
+In order to complete the bidirectional shutdown handshake, the peer needs
+to send back a "close notify" alert.
+The SSL_RECEIVED_SHUTDOWN flag will be set after receiving and processing
+it.
+SSL_shutdown() will return 1 when it has been received.
+
+The peer is still allowed to send data after receiving the "close notify"
+event.
+If the peer did send data it needs to be processed by calling SSL_read()
```

Comment:
If the peer may have sent data...

Reply:
I'm not sure what you're trying to say here. I think like it's worded now looks fine. It only needs to be done when it actually did send something. If it's possible that it did it, you clearly need to check it.

Reply:
Only wanted to point out that we can't tell if there is data pending.

Reply:
So does that mean you want to clarify it somehow? Like adding "It is recommended to call SSL_read() between SSL_shutdown() calls"?

Reply:
Yes, I guess I'd say something like that.